### ISSUE TITLE
Identity metadata

### ISSUE THREAD

**Issue by BeamerCola** [benedikt/layer-ruby#26]
I'm having an issue updating metadata on a user `Identity` object.

The error is:
```
{
  "url": "https://developer.layer.com/docs/platform",
  "data": {
    "property": "metadata"
  },
  "code": 9,
  "id": "invalid_operation",
  "message": "Patch operation \"set\" on \"metadata\" is not allowed."
}
```

Talking to Layer team they said
> yeah, you can only update or delete individual keys/values in identity metadata currently
> In other words: you CAN update metadata, just don’t write the entire object

Is this something that's supported?  Conversation has methods for metadata, but Identity didn't

Thanks!

**Comment by benedikt** [comment]
Hi @BeamerCola!

Thanks so much for getting in touch. It is possible to update metadata on identity objects. I'm not quite sure why it doesn't work for you, though. Can you please share the code snippet you use to update the metadata?

Cheers,

Benedikt

**Comment by BeamerCola** [comment]
Great thanks!

So by default `identity.metadata` returns nil
so setting `identity.metadata = {a: "b"}; identity.save`

It sounds like it's writing the whole metadata object and not patching existing? 

Thanks again, lemme know if there's anything else I can provide

**Comment by BeamerCola** [comment]
Keep messing with it and can't seem to find anything that works

**Comment by benedikt** [comment]
@BeamerCola You're right. It's a bug. I'll see what I can do to fix it.

**Comment by benedikt** [comment]
Hi @BeamerCola,

I just pushed a potential fix to the `identity-metadata` branch. Can you please check if it fixes your issue?

Use this line in your Gemfile to use the version from the branch:

```
gem 'layer-ruby', github: 'benedikt/layer-ruby', branch: 'identity-metadata'
```

Afterwards, these operations should work as expected:

```
identity.metadata = { foo: 'bar' }
identity.metadata['bar'] = 'baz'
```

Replacing the whole object will not work, though. I changed the `#metadata=` setter to simply set all key value pairs on the existing metadata object.

Please let me know if this fixes your issues.

**Comment by BeamerCola** [comment]
Works great!  You're the most attentive developer out there - keep up the awesome work.

Thank you.

**Comment by benedikt** [comment]
Thank you, @BeamerCola  😄 

I just released this fix as v0.7.2 on RubyGems. So you can now switch back to loading the gem from there.
